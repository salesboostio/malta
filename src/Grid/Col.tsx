import * as React from "react";
import * as classNames from "classnames";

export interface ColSize {
  span?: number;
  order?: number;
  offset?: number;
  push?: number;
  pull?: number;
}

export interface ColProps extends React.HTMLAttributes<HTMLDivElement> {
  span?: number;
  order?: number;
  offset?: number;
  push?: number;
  pull?: number;
  xs?: number | ColSize;
  sm?: number | ColSize;
  md?: number | ColSize;
  lg?: number | ColSize;
  xl?: number | ColSize;
  xxl?: number | ColSize;
  prefixCls?: string;
}

export class Col extends React.Component<ColProps, {}> {
  render() {
    const props: any = this.props;
    const { span, order, offset, push, pull, className, children, prefixCls = "col", ...others } = props;
    let sizeClassObj = {};
    ["xs", "sm", "md", "lg", "xl", "xxl"].forEach(size => {
      let sizeProps: ColSize = {};
      if (typeof props[size] === "number") {
        sizeProps.span = props[size];
      } else if (typeof props[size] === "object") {
        sizeProps = props[size] || {};
      }

      delete others[size];

      if (size === "xs") {
        sizeClassObj = {
          ...sizeClassObj,
          [`${prefixCls}-${sizeProps.span}`]: sizeProps.span !== undefined,
          [`order-${sizeProps.order}`]: sizeProps.order || sizeProps.order === 0,
          [`offset-${sizeProps.offset}`]: sizeProps.offset || sizeProps.offset === 0,
          [`push-${sizeProps.push}`]: sizeProps.push || sizeProps.push === 0,
          [`pull-${sizeProps.pull}`]: sizeProps.pull || sizeProps.pull === 0,
        };
      } else {
        sizeClassObj = {
          ...sizeClassObj,
          [`${prefixCls}-${size}-${sizeProps.span}`]: sizeProps.span !== undefined,
          [`order-${size}-${sizeProps.order}`]: sizeProps.order || sizeProps.order === 0,
          [`offset-${size}-${sizeProps.offset}`]: sizeProps.offset || sizeProps.offset === 0,
          [`push-${size}-${sizeProps.push}`]: sizeProps.push || sizeProps.push === 0,
          [`pull-${size}-${sizeProps.pull}`]: sizeProps.pull || sizeProps.pull === 0,
        };
      }
    });
    const classes = classNames({
      [`${prefixCls}-${span}`]: span !== undefined,
      [`order-${order}`]: order,
      [`offset-${offset}`]: offset,
      [`push-${push}`]: push,
      [`pull-${pull}`]: pull,
    }, className, sizeClassObj);

    return <div {...others} className={classes}>{children}</div>;
  }
}
