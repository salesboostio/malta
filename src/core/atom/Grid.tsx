import styled from "styled-components";
import { Block } from "./Block";
import {
  gridAutoColumns,
  GridAutoColumnsProps,
  gridAutoFlow,
  GridAutoFlowProps,
  gridAutoRows,
  GridAutoRowsProps,
  gridGap,
  GridGapProps,
  gridColumnGap,
  GridColumnGapProps,
  gridRowGap,
  GridRowGapProps,
  gridTemplateAreas,
  GridTemplatesAreasProps,
  gridTemplateColumns,
  GridTemplatesColumnsProps,
  gridTemplateRows,
  GridTemplatesRowsProps,
} from "styled-system";

export type GridProps =
  GridAutoColumnsProps & GridAutoFlowProps & GridAutoRowsProps
  & GridColumnGapProps & GridRowGapProps & GridTemplatesAreasProps
  & GridTemplatesColumnsProps & GridTemplatesRowsProps & GridGapProps;

export const Grid = styled(Block) <GridProps>`
  display: flex;
  ${gridAutoColumns}
  ${gridAutoFlow}
  ${gridAutoRows}
  ${gridColumnGap}
  ${gridRowGap}
  ${gridGap}
  ${gridTemplateAreas}
  ${gridTemplateColumns}
  ${gridTemplateRows}
`;
