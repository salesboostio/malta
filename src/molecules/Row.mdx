---
name: Row
menu: Molecules
---

import { Playground } from "docz"
import { App, Chip, Flex, Row } from "../../dist"

# Row

This is Row

## Dependencies
<App>
<Flex>
<Chip bg="blue.deep" color="#FFF">None</Chip>
</Flex>
</App>

## Properties

|Property|Type|Default|
|-|-|-|
|space|ResponsiveValue|0|

## Usecase

<Playground>
  <App>
    <Row />
  </App>
</Playground>

## Best Practices

## Test