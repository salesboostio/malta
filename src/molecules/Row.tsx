import * as React from "react";
import styled from "styled-components";
import { Flex } from "../atoms/Flex";

export const Row = styled(Flex)`
`;

Row.defaultProps = {
  flexDirection: "row",
  alignItems: "center",
  justifyContent: "center"
};
