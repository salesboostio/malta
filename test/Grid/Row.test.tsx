
import * as React from "react";
import { create } from "react-test-renderer";
import { Row } from "../../src/Grid/Row";

test("Row", () => {
    const node = create(<Row />).toJSON();
    expect(node).toMatchSnapshot();
});
