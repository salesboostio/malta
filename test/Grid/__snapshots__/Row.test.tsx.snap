// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row 1`] = `
<div
  className="row"
  style={undefined}
/>
`;
